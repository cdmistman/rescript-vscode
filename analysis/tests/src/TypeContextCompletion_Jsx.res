type someVariant = One | Two | Three | Four | Five(int) | Six(option<string>, int)

type otherVariant = [#one | #two | #three | #four | #five(int) | #six(option<int>, int)]

let someValue = Two

module SomeComponent = {
  @react.component
  let make = (
    ~someVariant: someVariant,
    ~anotherThing: TypeDefinition.variant,
    ~thirdThing: otherVariant,
  ) => {
    ignore(someVariant)
    ignore(anotherThing)
    ignore(thirdThing)
    React.null
  }
}

// let x = <SomeComponent someVaria
//                                 ^com

// let x = <SomeComponent someVariant=
//                                    ^com

// let x = <SomeComponent someVariant=
//                                     ^com

// let x = <SomeComponent someVariant=T
//                                     ^com

// let x = <SomeComponent someVariant=t
//                                     ^com

// let x = <SomeComponent anotherThing=
//                                     ^com

// let x = <SomeComponent thirdThing=
//                                   ^com

// let x = <SomeComponent thirdThing=
//                                    ^com

// let x = <SomeComponent thirdThing=#t
//                                     ^com

// let x = <SomeComponent thirdThing=#T
//                                     ^com
