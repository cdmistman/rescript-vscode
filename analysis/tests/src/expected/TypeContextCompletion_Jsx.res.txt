Complete src/TypeContextCompletion_Jsx.res 20:35
posCursor:[20:35] posNoWhite:[20:34] Found expr:[20:12->20:35]
JSX <SomeComponent:[20:12->20:25] someVaria[20:26->20:35]=...[20:26->20:35]> _children:None
Completable: Cjsx([SomeComponent], someVaria, [someVaria])
[{
    "label": "someVariant",
    "kind": 4,
    "tags": [],
    "detail": "someVariant",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Jsx.res 23:38
posCursor:[23:38] posNoWhite:[23:37] Found expr:[23:12->23:38]
JSX <SomeComponent:[23:12->23:25] someVariant[23:26->23:37]=...__ghost__[0:-1->0:-1]> _children:None
Found typed context
Completable: CtypedContext(JsxProp(<SomeComponent someVariant="" />))
Found variant type someVariant
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "Four",
    "kind": 4,
    "tags": [],
    "detail": "Four\n\n",
    "documentation": null
  }, {
    "label": "Five(_)",
    "kind": 4,
    "tags": [],
    "detail": "Five(int)\n\n",
    "documentation": null
  }, {
    "label": "Six(_)",
    "kind": 4,
    "tags": [],
    "detail": "Six(option<string>, int)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Jsx.res 26:39
posCursor:[26:39] posNoWhite:[26:37] Found expr:[26:12->26:38]
JSX <SomeComponent:[26:12->26:25] someVariant[26:26->26:37]=...__ghost__[0:-1->0:-1]> _children:None
Found typed context
Completable: CtypedContext(JsxProp(<SomeComponent someVariant="" />))
Found variant type someVariant
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "Four",
    "kind": 4,
    "tags": [],
    "detail": "Four\n\n",
    "documentation": null
  }, {
    "label": "Five(_)",
    "kind": 4,
    "tags": [],
    "detail": "Five(int)\n\n",
    "documentation": null
  }, {
    "label": "Six(_)",
    "kind": 4,
    "tags": [],
    "detail": "Six(option<string>, int)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Jsx.res 29:39
posCursor:[29:39] posNoWhite:[29:38] Found expr:[29:12->29:39]
JSX <SomeComponent:[29:12->29:25] someVariant[29:26->29:37]=...[29:38->29:39]> _children:None
Found typed context
Completable: CtypedContext(JsxProp(<SomeComponent someVariant=T />))
Found variant type someVariant
[{
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Jsx.res 32:39
posCursor:[32:39] posNoWhite:[32:38] Found expr:[32:12->32:39]
JSX <SomeComponent:[32:12->32:25] someVariant[32:26->32:37]=...[32:38->32:39]> _children:None
posCursor:[32:39] posNoWhite:[32:38] Found expr:[32:38->32:39]
Pexp_ident t:[32:38->32:39]
Completable: Cpath Value[t]
[]

Complete src/TypeContextCompletion_Jsx.res 35:39
posCursor:[35:39] posNoWhite:[35:38] Found expr:[35:12->35:39]
JSX <SomeComponent:[35:12->35:25] anotherThing[35:26->35:38]=...__ghost__[0:-1->0:-1]> _children:None
Found typed context
Completable: CtypedContext(JsxProp(<SomeComponent anotherThing="" />))
Found variant type TypeDefinition.variant
[{
    "label": "Foo",
    "kind": 4,
    "tags": [],
    "detail": "Foo\n\n",
    "documentation": null
  }, {
    "label": "Bar",
    "kind": 4,
    "tags": [],
    "detail": "Bar\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Jsx.res 38:37
posCursor:[38:37] posNoWhite:[38:36] Found expr:[38:12->38:37]
JSX <SomeComponent:[38:12->38:25] thirdThing[38:26->38:36]=...__ghost__[0:-1->0:-1]> _children:None
Found typed context
Completable: CtypedContext(JsxProp(<SomeComponent thirdThing="" />))
Found polyvariant type otherVariant
[{
    "label": "#one",
    "kind": 4,
    "tags": [],
    "detail": "#one",
    "documentation": null
  }, {
    "label": "#six(_)",
    "kind": 4,
    "tags": [],
    "detail": "#six(option<int>, int)",
    "documentation": null
  }, {
    "label": "#three",
    "kind": 4,
    "tags": [],
    "detail": "#three",
    "documentation": null
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two",
    "documentation": null
  }, {
    "label": "#five(_)",
    "kind": 4,
    "tags": [],
    "detail": "#five(int)",
    "documentation": null
  }, {
    "label": "#four",
    "kind": 4,
    "tags": [],
    "detail": "#four",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Jsx.res 41:38
posCursor:[41:38] posNoWhite:[41:36] Found expr:[41:12->41:37]
JSX <SomeComponent:[41:12->41:25] thirdThing[41:26->41:36]=...__ghost__[0:-1->0:-1]> _children:None
Found typed context
Completable: CtypedContext(JsxProp(<SomeComponent thirdThing="" />))
Found polyvariant type otherVariant
[{
    "label": "#one",
    "kind": 4,
    "tags": [],
    "detail": "#one",
    "documentation": null
  }, {
    "label": "#six(_)",
    "kind": 4,
    "tags": [],
    "detail": "#six(option<int>, int)",
    "documentation": null
  }, {
    "label": "#three",
    "kind": 4,
    "tags": [],
    "detail": "#three",
    "documentation": null
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two",
    "documentation": null
  }, {
    "label": "#five(_)",
    "kind": 4,
    "tags": [],
    "detail": "#five(int)",
    "documentation": null
  }, {
    "label": "#four",
    "kind": 4,
    "tags": [],
    "detail": "#four",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Jsx.res 44:39
posCursor:[44:39] posNoWhite:[44:38] Found expr:[44:12->44:39]
JSX <SomeComponent:[44:12->44:25] thirdThing[44:26->44:36]=...[44:37->44:39]> _children:None
Found typed context
Completable: CtypedContext(JsxProp(<SomeComponent thirdThing=t />))
Found polyvariant type otherVariant
[{
    "label": "#three",
    "kind": 4,
    "tags": [],
    "detail": "#three",
    "documentation": null
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Jsx.res 47:39
posCursor:[47:39] posNoWhite:[47:38] Found expr:[47:12->47:39]
JSX <SomeComponent:[47:12->47:25] thirdThing[47:26->47:36]=...[47:37->47:39]> _children:None
Found typed context
Completable: CtypedContext(JsxProp(<SomeComponent thirdThing=T />))
Found polyvariant type otherVariant
[]

